Source/Core/Common/Arm64Emitter.cpp

```diff
@@ -2003,96 +2004,152 @@ void ARM64XEmitter::ADRP(ARM64Reg Rd, s64 imm)
   EncodeAddressInst(1, Rd, static_cast<s32>(imm >> 12));
 }
 
-// Wrapper around MOVZ+MOVK (and later MOVN)
-void ARM64XEmitter::MOVI2R(ARM64Reg Rd, u64 imm)
+template <typename T, size_t MaxSize>
+class SmallVector final
 {
-  unsigned int parts = Is64Bit(Rd) ? 4 : 2;
-  BitSet32 upload_part(0);
+public:
+  SmallVector() {}
+  SmallVector(size_t size) : m_size(size) {}
```

Comment:
explicit

Reply:
This constructor should probably initialize the array member, otherwise uninitialized values can be returned from operator[]... Alternative: `std::array<T, MaxSize> m_array{};`